src/main/java/org/springframework/cloud/deployer/spi/kubernetes/DefaultContainerFactory.java

```diff
@@ -273,9 +273,16 @@ private ProbeType getProbeType(ContainerConfiguration containerConfiguration) {
 	 */
 	List<String> createCommandArgs(AppDeploymentRequest request) {
 		List<String> cmdArgs = new LinkedList<>();
-		List<String> commandArgOptions = request.getCommandlineArguments().stream()
-				.map(arg-> arg.substring(0,arg.indexOf("=")).replaceAll("^--",""))
```

Comment:
**[OPINION-NITatom]** I think leaving this as a simple stream+map but have the map use a method reference that hides all of the "how to get the arg" would be easier to follow. 

Something like: 
```java
List<String> commandArgOptions = request.getCommandlineArguments().stream()
    .map(this::getArgOption)
    .collect(Collectors.toList());
}

private String getArgOption(String arg) {
   int indexOfAssignment = arg.indexOf("=");
   String argOption = (indexOfAssignment < 0) ? arg : arg.substring(0, indexOfAssignment);
   return argOption.trim().replaceAll("^--", "");
}
```